src/test/util/setup_common.cpp

```diff
@@ -97,9 +101,9 @@ BasicTestingSetup::BasicTestingSetup(const std::string& chainName, const std::ve
     {
         SetupServerArgs(*m_node.args);
         std::string error;
-        const bool success{m_node.args->ParseParameters(arguments.size(), arguments.data(), error)};
-        assert(success);
-        assert(error.empty());
+        if (!m_node.args->ParseParameters(arguments.size(), arguments.data(), error)) {
+            throw std::runtime_error{error};
```

Comment:
Is it now impossible to use DEBUG_LOG_OUT? If yes, might as well remove it.

Reply:
No, it is not broken by this PR. For example, both commands below produce the same output:

```sh
test_bitcoin --run_test="net_tests/cnode_listen_port" -- DEBUG_LOG_OUT
test_bitcoin --run_test="net_tests/cnode_listen_port" -- -printtoconsole=1
```